pkg/tee/aes.go

```diff
@@ -0,0 +1,62 @@
+package tee
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+)
+
+// Encrypt encrypts a plain text using AES-GCM.
+func EncryptAES(plainText, key string) (string, error) {
+	block, err := aes.NewCipher([]byte(key))
+	if err != nil {
```

Comment:
Minor: in all of this file, can we enrich the errors a bit to provide more context as to when/where they happened?